server/services/placeOnReportService.ts

```diff
@@ -266,14 +269,46 @@ export default class PlaceOnReportService {
     const manageAdjudicationsClient = new ManageAdjudicationsClient(user.token)
     const { draftAdjudication } = await manageAdjudicationsClient.getDraftAdjudication(draftAdjudicationId)
 
+    const statementComplete = draftAdjudication.incidentStatement?.completed || false
+    const offenceDetailsComplete = this.checkOffenceDetails(draftAdjudication.offenceDetails)
+    const offenceDetailsStatus = this.getOffenceDetailsStatus(offenceDetailsComplete)
+
+    const offenceDetailsUrl = this.getNextOffencesUrl(draftAdjudication.offenceDetails, draftAdjudication.id)
+    const incidentStatementStatus = this.getIncidentStatementStatus(
+      !!draftAdjudication.incidentStatement,
+      statementComplete
+    )
+
     return {
+      offenceDetailsUrl,
       handoverDeadline: draftAdjudication.incidentDetails.handoverDeadline,
-      statementPresent: !!draftAdjudication.incidentStatement,
-      statementComplete: draftAdjudication.incidentStatement?.completed || false,
-      offenceDetailsComplete: draftAdjudication.offenceDetails?.length > 0,
+      incidentStatementStatus,
+      offenceDetailsStatus,
+      showLinkForAcceptDetails: offenceDetailsComplete && statementComplete,
     }
   }
 
+  checkOffenceDetails(offenceDetails: OffenceDetails[]): boolean {
+    return offenceDetails?.length > 0 || false
+  }
+
+  getNextOffencesUrl(offenceDetails: OffenceDetails[], adjudicationId: number): string {
+    const offenceDetailsComplete = this.checkOffenceDetails(offenceDetails)
+    if (offenceDetailsComplete) return adjudicationUrls.detailsOfOffence.urls.start(adjudicationId)
+    return adjudicationUrls.ageOfPrisoner.urls.start(adjudicationId)
+  }
+
+  getIncidentStatementStatus = (statementPresent: boolean, statementComplete: boolean): IncidentStatementStatus => {
+    if (!statementPresent) return { classes: 'govuk-tag govuk-tag--grey', text: 'NOT STARTED' }
```

Comment:
Well this is still some presentation logic in the service layer, which I am keen to avoid.

But it's ok for now.